uniprotpy/protein_tree.py

```diff
@@ -16,11 +16,11 @@ def create_protein_tree(proteome):
   proteins = list(SeqIO.parse(proteome, 'fasta'))
 
   # get UniProt IDs for one protein per gene proteome
-  gp_ids = [str(x.id.split('|')[1]) for x in list(SeqIO.parse(gp_proteome, 'fasta'))]
+  # TODO: See why this was gp_proteome rather than proteome
+  gp_ids = [str(x.id.split('|')[1]) for x in list(SeqIO.parse(proteome, 'fasta'))]
```

Comment:
This needs to be `gp_proteome` - a feature of UniProt is the separation of what I call the "gene priority proteome" and the regular proteome. The gene priority is UniProt curated selection of the best representative protein for a gene. Since humans have ~20,000 genes, but many different isoforms, there is a file downloadable on UniProt that picks the best isoform for each gene. We use this subset for all kinds of things.

There might be a more elegant way of distinguishing these isoforms from the rest. But for now, I just grab the UniProt IDs that are in the gene priority proteome file.

Reply:
I changed this because my IDE bitched about gp_proteome not existing - where is it defined? Or will it be defined in the future?

Reply:
Oh, must've been an oversight. Woops, we'll need to add that as an option. `protein_tree.py` is sort of not really up to date anyways, I have changes to make, we can exclude that for now.